activation.cpp

```diff
@@ -106,10 +143,31 @@ auto Activation::activation(Activations value) -> Activations
         using UpdateAccessKey = phosphor::software::image::UpdateAccessKey;
         UpdateAccessKey updateAccessKey(manifestPath);
 
+        using VersionClass = phosphor::software::manager::Version;
+
+        std::string versionID =
+            VersionClass::getValue(manifestPath.string(), "version");
+        std::string currVersion = versionID.substr(2, 4);
+        std::string buildIDTrunc = getBuildID(manifestPath);
```

Comment:
we can do "getValue() for BuildId here and do the regex compare instead of having a getBuildID function.